tests/contrib/pymongo/test.py

```diff
@@ -461,7 +471,74 @@ def test_user_specified_service(self):
         tracer = get_dummy_tracer()
         client = pymongo.MongoClient(port=MONGO_CONFIG["port"])
         Pin.get_from(client).clone(tracer=tracer).onto(client)
+        # We do not wish to trace tcp spans here
+        Pin.get_from(pymongo.server.Server).remove_from(pymongo.server.Server)
         client["testdb"].drop_collection("whatever")
         spans = tracer.writer.pop()
         assert len(spans) == 1
         assert spans[0].service != "mysvc"
+
+
+class TestPymongoSocketTracing(BaseTracerTestCase):
+    """
+    Test suite which checks that tcp socket creation/retrieval is correctly traced
+    """
+    def setUp(self):
+        patch()
+        self.tracer = get_dummy_tracer()
+        # Override server pin's tracer with our dummy tracer
+        Pin.override(pymongo.server.Server, tracer=self.tracer)
+        # maxPoolSize controls the number of sockets that the client can instanciate
+        # and choose from to perform classic operations. For the sake of our tests,
+        # let's limit this number to 1
+        self.client = pymongo.MongoClient(port=MONGO_CONFIG['port'], maxPoolSize=1)
+        # Override TracedMongoClient's pin's tracer with our dummy tracer
+        Pin.override(self.client, tracer=self.tracer)
+
+    def tearDown(self):
+        unpatch()
+        self.client.close()
+
+    @staticmethod
+    def check_socket_metadata(span):
+        assert span.name == "pymongo.get_socket"
+        assert span.service == mongox.SERVICE
+        assert span.span_type == SpanTypes.MONGODB.value
+        assert span.meta.get('out.host') == "localhost"
+        assert span.metrics.get('out.port') == MONGO_CONFIG["port"]
+
+    def test_single_op(self):
+        self.client["some_db"].drop_collection("some_collection")
+        spans = self.tracer.writer.pop()
+
+        assert len(spans) == 2
+        self.check_socket_metadata(spans[0])
+        assert spans[1].name == "pymongo.cmd"
+
+    def test_multiple_ops(self):
+        input_movies = [
+            {'name': 'Kool movie', 'filmmaker': 'John Mc Johnson'},
+        ]
+        db = self.client["medias"]
+        db.movies.insert_many(input_movies)
```

Comment:
We should also verify that the core functionality of pymongo is unaffected by our changes. I think you can just use the tests you have and then make sure that the operations worked as expected 🙂

Reply:
You're right! Pushed that :)

Reply:
Looks like it breaks some tests, that was definitely a good idea! Gonna look into that